Better integration with Tasmota features

sfromis opened the pull request:
Using the driver with a TMP117 sensor, I observed how the driver did not support features TempOffset, TempRes and SetOption8 (fahrenheit). Added this support, including allowing changes while running. Also made a couple of tweaks to take advantage of newer Tasmota features. 

Another reason to avoid the global `SENSOR_ADDR` is that this could clash with other drivers using the same name. While this could have been handled prefixing the name of the global variable, the new feature of static variables is very nice for such fixed setup. Also made the temperature translation static, observed how the existing driver used the name "Temperatur", which fits some languages only :-)

sfromis committed:
Better integration with Tasmota features

Using the driver with a TMP117 sensor, I observed how the driver did not support features TempOffset, TempRes and SetOption8 (fahrenheit). Added this support, including allowing changes while running. Also made a couple of tweaks to take advantage of newer Tasmota features. 

Another reason to avoid the global `SENSOR_ADDR` is that this could clash with other drivers using the same name. While this could have been handled prefixing the name of the global variable, the new feature of static variables is very nice for such fixed setup. Also made the temperature translation static, observed how the existing driver used the name "Temperatur", which fits some languages only :-)

TomsTek commented:
Your changes look great, but i have a little problem. My Tasmota (official build: (Tasmota Version | 10.1.0(tasmota)) create cyclic exceptions.
```
BRY: Exception> 'attribute_error' - class 'TMP117' has no method 'temp_format'
stack traceback:
	<unknown source>: in function `event`
```

Did i need to start the driver in a special  way or link additional modules? 
I start the driver with 'load("tmp117.be")'

First messages after startup:
```
load("tmp117.be")
I2C: TMP117 detected on bus 1
TMP117 TempRes:2 TempOffset:0 Fahrenheit:false
true
TMP117 TempRes:1 TempOffset:0 Fahrenheit:false
BRY: Exception> 'attribute_error' - class 'TMP117' has no method 'temp_format'
stack traceback:
	<unknown source>: in function `event`
TMP117 TempRes:1 TempOffset:0 Fahrenheit:false
BRY: Exception> 'attribute_error' - class 'TMP117' has no method 'temp_format'
stack traceback:
	<unknown source>: in function `event`
BRY: Exception> 'attribute_error' - class 'TMP117' has no method 'temp_format'
```

sfromis committed:
Fix compatibility with previous version

You're right, sorry. I made and tested the changes on latest development version (2022.01.2, new version numbering system), and did not take into account that an old issue in Berry was fixed, with only local variables usable for functions calls, instead of straight away using `self.temp_format` to make the call. Changed to copy the function reference into a local variable; the updated driver should work with current release version, of course.

sfromis committed:
A bit of finishing edits fell out, redone

Reinstated consistency in formatting.

sfromis commented:
Should work better now, at least with current release version. I was not sure whether to leave the debugging message printed when the temperature formatting function is recreated, but saw no harm in such a low number of chatter.

TomsTek commented:
With the new changes the driver work fine. Great work ;)
